java/vector/src/main/java/org/apache/arrow/vector/complex/AbstractStructVector.java

```diff
@@ -196,21 +245,51 @@ protected void putChild(String name, FieldVector vector) {
     putVector(name, vector);
   }
 
+  private void put(String name, FieldVector vector, boolean overwrite) {
+    final boolean old = vectors.put(
+        Preconditions.checkNotNull(name, "field name cannot be null"),
+        Preconditions.checkNotNull(vector, "vector cannot be null"),
+        overwrite
+    );
+    if (old) {
+      logger.debug("Field [{}] mutated to [{}] ", name,
+          vector.getClass().getSimpleName());
+    }
+  }
+
   /**
    * Inserts the input vector into the map if it does not exist, replaces if it exists already.
```

Comment:
This docstring no longer matches the behavior.

Reply:
fixed, thanks!